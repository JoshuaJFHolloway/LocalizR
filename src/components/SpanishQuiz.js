import React, { Component } from 'react';
import SpanishQuizQ from './SpanishQuizQ';


class SpanishQuiz extends Component {
  constructor() {
    super();
    this.state = {
      score: 0,
      counter: 0,
      nextQuestion: false,
      lastQuestion: false,
    }
  }


  handleCorrectAnswerSubmitted = () => {
    this.setState({
      counter: this.state.counter + 1,
      score: this.state.score + 1
    })
  };



  handleAnswerSubmitted = () => {
    this.setState({
      counter: this.state.counter + 1
    })
  };


  decider = () => {

  };



  render() {
    return (
      <SpanishQuizQ
        handleAnswerSubmitted = {this.handleAnswerSubmitted}
        handleCorrectAnswerSubmitted = {this.handleCorrectAnswerSubmitted}
      />
    )
  }
}

export default SpanishQuiz;