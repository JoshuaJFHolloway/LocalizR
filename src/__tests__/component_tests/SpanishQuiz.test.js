import React from 'react';
import { shallow } from 'enzyme';
import SpanishQuiz from '../../components/SpanishQuiz';

describe('SpanishQuiz', () => {
  let spanishquiz = shallow(<SpanishQuiz />);

  describe('Renders components', () => {
    it('renders correctly', () => {
      expect(spanishquiz).toMatchSnapshot();
    });

    it('renders SpanishQuizQ page', () => {
      expect(spanishquiz.find('SpanishQuizQ').exists()).toBe(true);
    });
  });
});
