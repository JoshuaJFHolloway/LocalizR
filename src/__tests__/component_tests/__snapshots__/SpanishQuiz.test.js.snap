// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpanishQuiz Renders components renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SpanishQuiz />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <SpanishQuizQ
        answer1="el vestido rojo"
        answer2="Los pantalones rojos"
        answer3="La camiseta roja"
        correctAnswer="el traje rojo"
        handleAnswerSubmitted={[Function]}
        handleCorrectAnswerSubmitted={[Function]}
        picture="https://www.opposuits.co.uk/media/catalog/product/cache/1/image/550x/925f46717e92fbc24a8e2d03b22927e1/s/s/ss_reddevil-lifestyle-01.jpg"
        question="Which of these is your suit?"
        scenario="You choose to wear this for your date"
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "answer1": "el vestido rojo",
        "answer2": "Los pantalones rojos",
        "answer3": "La camiseta roja",
        "correctAnswer": "el traje rojo",
        "handleAnswerSubmitted": [Function],
        "handleCorrectAnswerSubmitted": [Function],
        "picture": "https://www.opposuits.co.uk/media/catalog/product/cache/1/image/550x/925f46717e92fbc24a8e2d03b22927e1/s/s/ss_reddevil-lifestyle-01.jpg",
        "question": "Which of these is your suit?",
        "scenario": "You choose to wear this for your date",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <SpanishQuizQ
          answer1="el vestido rojo"
          answer2="Los pantalones rojos"
          answer3="La camiseta roja"
          correctAnswer="el traje rojo"
          handleAnswerSubmitted={[Function]}
          handleCorrectAnswerSubmitted={[Function]}
          picture="https://www.opposuits.co.uk/media/catalog/product/cache/1/image/550x/925f46717e92fbc24a8e2d03b22927e1/s/s/ss_reddevil-lifestyle-01.jpg"
          question="Which of these is your suit?"
          scenario="You choose to wear this for your date"
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "answer1": "el vestido rojo",
          "answer2": "Los pantalones rojos",
          "answer3": "La camiseta roja",
          "correctAnswer": "el traje rojo",
          "handleAnswerSubmitted": [Function],
          "handleCorrectAnswerSubmitted": [Function],
          "picture": "https://www.opposuits.co.uk/media/catalog/product/cache/1/image/550x/925f46717e92fbc24a8e2d03b22927e1/s/s/ss_reddevil-lifestyle-01.jpg",
          "question": "Which of these is your suit?",
          "scenario": "You choose to wear this for your date",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
